clippy_lints/src/manual_checked_sub.rs

```diff
@@ -0,0 +1,239 @@
+        if !self.msrv.meets(cx, msrvs::MANUAL_CHECKED_SUB) {
+            return;
+        }
+
+        let applicability = Applicability::MachineApplicable;
+
+        if let ExprKind::If(drop_temp_expr, body_block, else_block) = expr.kind
+            && let ExprKind::DropTemps(condition_expr) = drop_temp_expr.kind
+            && let ExprKind::Binary(op, lhs, rhs) = condition_expr.kind
+            && is_unsigned_int(cx, lhs)
+            && is_unsigned_int(cx, rhs)
+        {
+            if let BinOpKind::Ge = op.node {
+                SubExprVisitor {
+                    cx,
+                    if_expr: expr,
+                    if_body_block: body_block,
+                    else_block,
+                    applicability,
+                    condition_lhs: lhs,
+                    condition_rhs: rhs,
+                }
+                .visit_expr(body_block);
+            }
+
+            if let BinOpKind::Gt = op.node {
+                if let ExprKind::Lit(lit) = &rhs.kind
+                    && let LitKind::Int(Pu128(0), _) = lit.node
+                {
+                    SubExprVisitor {
+                        cx,
+                        if_expr: expr,
+                        if_body_block: body_block,
+                        else_block,
+                        applicability,
+                        condition_lhs: lhs,
+                        condition_rhs: rhs,
+                    }
+                    .visit_expr(body_block);
+                }
+            }
+        }
+    }
+}
+
+struct SubExprVisitor<'a, 'tcx> {
+    cx: &'a LateContext<'tcx>,
+    if_expr: &'tcx Expr<'tcx>,
+    if_body_block: &'tcx Expr<'tcx>,
+    else_block: Option<&'tcx Expr<'tcx>>,
+    applicability: Applicability,
+    condition_lhs: &'tcx Expr<'tcx>,
+    condition_rhs: &'tcx Expr<'tcx>,
+}
+
+impl<'tcx> Visitor<'tcx> for SubExprVisitor<'_, 'tcx> {
+    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) {
+        if let ExprKind::Binary(op, sub_lhs, sub_rhs) = expr.kind
+            && let BinOpKind::Sub = op.node
+        {
+            if let ExprKind::Lit(lit) = self.condition_rhs.kind
+                && let LitKind::Int(Pu128(0), _) = lit.node
+                && (is_referencing_same_variable(sub_lhs, self.condition_lhs)
+                    || are_literals_equal(sub_lhs, self.condition_lhs))
+            {
+                self.build_suggession(expr);
+            } else if (is_referencing_same_variable(sub_lhs, self.condition_lhs)
+                || are_literals_equal(sub_lhs, self.condition_lhs))
+                && (is_referencing_same_variable(sub_rhs, self.condition_rhs)
+                    || are_literals_equal(sub_rhs, self.condition_rhs))
+            {
+                self.build_suggession(expr);
+            }
+        }
+
+        walk_expr(self, expr);
+    }
+}
+
+impl<'tcx> SubExprVisitor<'_, 'tcx> {
+    fn build_suggession(&mut self, expr: &Expr<'tcx>) {
```

Comment:
The name has several issues:
- Typo (`suggession` → `suggestion`)
- It doesn't build anything (a builder usually returns a value).

Maybe `emit_lint()` would be a better name.